crates/autopilot/src/solvable_orders.rs

```diff
@@ -683,17 +683,15 @@ async fn filter_unsupported_tokens(
     .flatten()
     .collect::<HashSet<_>>();
 
-    let mut index = 0;
-    'outer: while index < orders.len() {
-        for token in orders[index].data.token_pair().unwrap() {
-            if bad_tokens.contains(&token) {
-                orders.swap_remove(index);
-                continue 'outer;
-            }
-        }
-        index += 1;
-    }
-    Ok(orders)
+    orders.retain(|order| {
+        !order
+            .data
+            .token_pair()
+            .unwrap_or_default()
+            .into_iter()
+            .any(|token| bad_tokens.contains(&token))
```

Comment:
nit: Nothing that will cause problems in practice but there is a subtle issue with `.unwrap_or_default()` here. In case an order sells and buys the same token this would cause the loop to think the traded tokens are actually `0x00000` and `0x00001` due to the weird `Default` implementation for testing purposes.
With `into_iter().flatten()` this would actually result in 0 tokens being checked.
Also inverting the logic to use `.all(|token| !bad_tokens.contains(&token))` seems a bit easier to understand IMO.
Note, that switching to `into_iter().flatten()` also requires the switch to use `.all()` due to the different default behavior of `any()` and `all()` on empty iterators.

```suggestion
        order
            .data
            .token_pair()
            .into_iter()
            .flatten()
            .all(|token| !bad_tokens.contains(&token))
```